### ISSUE TITLE
Run checks to use latest supported Python (3.9)

### ISSUE THREAD

**Issue by plamut** [googleapis/python-bigquery#462]
In `setup.py` and in README we declare that we support `Python >= 3.6`, but the nox test sessions only use Python up to 3.8. Python 3.9 is also missing from the `classifiers` list in `setup.py`.

We should bump the maximum versions to `3.9` where applicable. If tests fail in 3.9, we should bound the `python_requires` version in `setup.py` until the failures are fixed (though - bound it in any case and manually increase only after we confirm that it works with the new minor Python release?).

If we first need to setup Kokoro environment to support Python 3.9 test environment, let's do that and keep this ticket open as a reminder.

cc: @tswast @busunkim96

**Comment by tswast** [comment]
This was blocked by grpcio and arrow have been released for both platforms now 
(https://github.com/grpc/grpc/issues/24344#issuecomment-725303543, https://issues.apache.org/jira/browse/ARROW-10224)

**Comment by plamut** [comment]
As a quick check, I tried to install the library with Python 3.9, but failed, because `pyarrow` pulled in `numpy==1.16.0`, but the latter failed to compile on my machine (Ubuntu), and there is no wheel for it. It didn't help if I installed `numpy==1.19.5`, pyarrow was still stubbornly pulling in `1.16.0`.

TL; DR - I was not able to install BigQuery localy with Python3.9, and thus could not run the tests either.

**Comment by tswast** [comment]
I misunderstood the ticket. They have released 3.9 wheels to nightly but not public PyPI yet. See my thread talking to myself in: https://github.com/googleapis/python-bigquery-storage/pull/99

**Comment by tswast** [comment]
Arrow 3.0 release with Python 3.9 support is in the works. https://github.com/apache/arrow/pull/9245

**Comment by kszucs** [comment]
The wheels are available now on pypi.

**Comment by plamut** [comment]
@kszucs That's great, thanks a lot!

**Comment by galuszkak** [comment]
@plamut I think this ticket can be closed.
